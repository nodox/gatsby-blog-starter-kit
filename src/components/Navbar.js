import React from 'react';
import Link from 'gatsby-link';
import Helmet from 'react-helmet';
import classNames from 'classnames';

export default class Navbar extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    const {
      currentPath,
      navLinks,
      config,
    } = this.props;

    const {
      pages,
      socialLinks,
    } = config;

    const links = navLinks.map((name, idx) => {
      const obj = pages[name];

      return (
        <li key={idx} className={obj.path === currentPath ? 'active' : ''}>
          <Link to={obj.path}>{obj.title}</Link>
        </li>
      );
    });

    const socialMediaLinks = socialLinks.map((obj, idx) => {
      return (
        <li key={idx}>
          <a href={obj.path} className={`icon ${obj.icon}`}>
            <span className="label">{obj.name}</span>
          </a>
        </li>
      );
    });

    return (
      <nav id="nav">
        <ul className="links">
          {links}
        </ul>
        <ul className="icons">
          {socialMediaLinks}
        </ul>
      </nav>
    );
  }
}
