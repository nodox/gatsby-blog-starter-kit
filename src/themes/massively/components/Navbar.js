import React from 'react';
import Link from 'gatsby-link';
import Helmet from 'react-helmet';
import classNames from 'classnames';


// <GatsbyLink to={post.frontmatter.path}>
//   {post.frontmatter.title}
// </GatsbyLink>

// <Helmet
//   title="Gatsby Default (Blog) Starter"
//   meta={[
//     { name: 'description', content: 'Sample' },
//     { name: 'keywords', content: 'sample, something' },
//   ]}
// />
export default class Navbar extends React.Component {
  constructor(props) {
    super(props)

    this.paths = [
      {
        path: '/',
        title: 'Home',
      },
      {
        path: '/about',
        title: 'About'
      }, 
      {
        path: '/speaking',
        title: 'Speaking'
      }, 
      {
        path: '/contact',
        title: 'Contact'
      },
    ]

    this.socialLinks = [
      {
        path: 'https://www.twitter.com/stevennatera',
        icon: 'twitter',
        name: 'Twitter',
      },
      {
        path: 'https://www.linkedin.com/in/snatera',
        icon: 'linkedin',
        name: 'LinkedIn',
      },
      {
        path: 'https://www.github.com/nodox',
        icon: 'github',
        name: 'Github',
      }
    ]
  }

  render() {

    const isActiveTab = (idx) => {
      let activeTab = classNames({
        'active': this.paths[idx].path === this.props.currentPath
      });
      return activeTab;
    };

    const links = this.paths.map((obj, idx) => {
      return (
        <li key={idx} className={isActiveTab(idx)}>
          <Link to={obj.path}>{obj.title}</Link>
        </li>
      );
    }); 

    const socialLinks = this.socialLinks.map((obj, idx) => {
      return (
        <li key={idx}>
          <a href={obj.path} className={`icon fa-${obj.icon}`}><span className="label">{obj.name}</span></a>
        </li>
      );
    });

    return (
      <nav id="nav">
        <ul className="links">
          {links}
        </ul>
        <ul className="icons">
          {socialLinks}
        </ul>
      </nav>
    );
  }

}