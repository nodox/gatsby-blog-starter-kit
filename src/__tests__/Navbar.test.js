import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';
import config from '../theme-config';

describe('<Navbar />', () => {
  const navLinks = [];
  const currentPath = '/hello';

  const component = shallow(
    <Navbar
      navLinks={navLinks}
      config={config}
      currentPath={currentPath} />
  );

  const props = component.instance().props;

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have the right props', () => {
    expect(props).toHaveProperty('config');
    expect(props).toHaveProperty('config.pages');
    expect(props).toHaveProperty('config.socialLinks');
    expect(props).toHaveProperty('navLinks');
    expect(props).toHaveProperty('currentPath');
  });

});
