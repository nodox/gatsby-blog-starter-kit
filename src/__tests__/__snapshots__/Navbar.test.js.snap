// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render correctly 1`] = `
<nav
  id="nav"
>
  <ul
    className="links"
  />
  <ul
    className="icons"
  >
    <li
      key="0"
    >
      <a
        className="icon fa-twitter"
        href="https://www.twitter.com/stevennatera"
      >
        <span
          className="label"
        >
          Twitter
        </span>
      </a>
    </li>
    <li
      key="1"
    >
      <a
        className="icon fa-linkedin"
        href="https://www.linkedin.com/in/snatera"
      >
        <span
          className="label"
        >
          LinkedIn
        </span>
      </a>
    </li>
    <li
      key="2"
    >
      <a
        className="icon fa-github"
        href="https://www.github.com/nodox"
      >
        <span
          className="label"
        >
          Github
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
